packages/web3-core/src/web3_request_manager.ts

```diff
@@ -438,15 +437,6 @@ export class Web3RequestManager<
 			throw new ResponseError(response, 'Got batch response for a normal request.');
 		}
 
-		if (
-			(jsonRpc.isResponseWithError(response) || jsonRpc.isResponseWithResult(response)) &&
```

Comment:
This conditional can never happen as we check `jsonRpc.isResponseWithError(response)` and  ` jsonRpc.isResponseWithResult(response)` seperately